client/question/components/LearnerViewDirective.js

```diff
@@ -780,6 +838,8 @@ tie.directive('learnerView', [function() {
             question = QuestionDataService.getQuestion(questionId);
             initQuestionData(questionId);
           }
+          $scope.speechBalloonsList =
```

Comment:
I don't believe this is used. Delete?

Reply:
This is used in speechBalloonsContainer directive. The existing code calls getSpeechBalloonsList() in ng-repeat, but this results in repainting of the balloons, making animation very difficult. However, simply assigning speechBalloonsList in the speechBalloonsContainer controller breaks the code reset button, as this approach does not handle removal of items during reset. So, speechBalloonsList is set in the learnerView directive whenever the user clicks on reset code.

Reply:
OK, changed to option 2. Thanks seanlip for the tips on how exactly how to implement it.

Reply:
So we should still delete this and the invocation below, right (and possibly even `$scope.ConversationLogDataService = ConversationLogDataService;` above)?

Reply:
Oh sorry. Didn't realize you hadn't pushed commits yet.